install/pre-install-payload/scripts/reqs-deploy.sh

```diff
@@ -183,7 +183,23 @@ EOF
 		sed -i -e "1s|^|imports = [\"${containerd_imports_path}/nydus-snapshotter.toml\"]\n|" ${containerd_config}
 	fi
 
-	sed -i -e "s|disable_snapshot_annotations = true|disable_snapshot_annotations = false|" ${containerd_config}
+	# Annotations should be passed down to the remote snapshotter in order to
+	# make it work. This can be done by setting `disable_snapshot_annotations = false`
+	# in the containerd's config.toml.
+	if grep -q 'disable_snapshot_annotations' "$containerd_config"; then
+		sed -i -e "s|disable_snapshot_annotations = true|disable_snapshot_annotations = false|" \
+			"${containerd_config}"
+	else
+		# In case the property does not exist, let's append it to the
+		# [plugins."io.containerd.grpc.v1.cri".containerd] section.
+		sed -i '/\[plugins\..*\.containerd\]/adisable_snapshot_annotations = false' \
```

Comment:
is this a typo? `adisable_snapshot_annotations` vs. `disable_snapshot_annotations`?

Reply:
@portersrc the next char after '/' can be a command to sed. In this case, '/a' tells to Append the following string (i.e disable_snapshot_annotations) ;)

Thanks for the careful review!

Reply:
fwiw: if we want to keep the indent this should work w/ nested capture groups:

```bash
sed -e 's|\(\(\s*\)\[plugins.\"io.containerd.grpc.v1.cri\".containerd\]\)|\1\n\2  ohai=true|' /etc/containerd/config.toml | grep -C1 ohai
    [plugins."io.containerd.grpc.v1.cri".containerd]
      ohai=true
      default_runtime_name = "runc"
 ```